### DISCUSSION TITLE
Training Arabic+English word detection model

### DISCUSSION THREAD

**Discussion by akshat311** [mindee/doctr#1448] [General]
I am trying to train a word detection model on a synthetically generated dataset having 10k documents. The document has English and Arabic words. I am using the following command to train 

`python references/detection/train_pytorch.py ./doctr_dataset_feb1/train ./doctr_dataset_feb1/test db_resnet50 --epochs 5 --pretrained --wb --workers 8`

However, the validation loss is not reducing below 0.8. Mean IOU also reaches a max of 0.75
Can someone suggest any hyperparameter/augmentation changes that can improve the results.
The image dimensions in the dataset are very different from each other. Can that be a problem?

**Comment by akshat311** [comment]
@felixT2K

**Reply by felixdittrich92** [reply]
try to change the lr scheduler to polylr (there is also already an open PR) see: https://github.com/mindee/doctr/pull/1444

**Reply by felixdittrich92** [reply]
btw a value of 75% IoU is already pretty good how looks precision and recall ?

**Reply by akshat311** [reply]
precision and recall are around 92%

**Reply by felixdittrich92** [reply]
this is mostly perfect xD ^^

**Reply by akshat311** [reply]
The results are still not up to the required accuracy. Are 10k samples not enough? Or should I train for much more epochs?

**Reply by akshat311** [reply]
on small images results are good. But on large images, still not very accurate. Can I try changing the default image size?

**Reply by felixdittrich92** [reply]
You should let it run until the loss does not decrease over 4 - 5 epochs (so set a higher number of epochs)
If it already works on small images but not on larger ones it makes in my mind no sense to lower the image size (currently padded with keeping aspect ratio to 1024x1024)
I would give the polylr scheduler a try (and check that also such larger samples are in your validation data) :)

**Reply by felixdittrich92** [reply]
@akshat311 works ? ^^
Btw we are always happy if users share there models with the community would you be interested ? :)
